openapi_tester/schema_tester.py

```diff
@@ -295,8 +295,7 @@ def test_openapi_object(
         required_keys = [key for key in schema_section.get("required", []) if key not in write_only_properties]
         response_keys = data.keys()
         additional_properties: Optional[Union[bool, dict]] = schema_section.get("additionalProperties")
-        if not properties and isinstance(additional_properties, dict):
-            properties = additional_properties
+        additional_properties_allowed = additional_properties is not None
```

Comment:
Since we specify 

```python
elif isinstance(additional_properties, dict):
```

we currently won't have any handling for non-bool/non-dict types. What do you think about amending this assignment to this, so all data types are handled?

```suggestion
        additional_properties_allowed = additional_properties is True or isinstance(additional_properties, dict)
```

Reply:
`additionalProperties` can only be a boolean or an object, so I think those two would be roughly equivalent. The only difference, if the schema is not considered invalid by the schema validation, would be in how we want to handle schemas where `additionalProperties` is defined as a different type.

So say there's a schema defined like this:
```
type: object
additionalProperties: 1
```

Do we want to allow `additionalProperties` in this case or not? With the `is not None` check we would allow them but not validate anything about them. With your proposed change we would not allow `additionalProperties`. I don't mind either option.

Reply:
From what I can gather, it would be possible for users to specify whatever they want, but anything other than a dict/bool/null would be a violation of the spec - does that seem right? If that's correct, we're deciding whether we think it's within the scope of this library to raise an error for the invalidity of the schema or not.

I can also see both sides, but am probably leaning towards raising an error here. Perhaps instead of the suggestion I made above, we should do something like this?

```python
if not isinstance(additional_properties, (bool, dict, None)):
    raise OpenAPISchemaError('Invalid additional properties type')
additional_properties_allowed = additional_properties is not None
```

What do you think @Goldziher? 🙂

Reply:
yes, i agree @sondrelg - i think your last suggestion is the cleanest of what has been suggested so far.

Reply:
I've added the check and a test. Changed it a little bit because `None` cannot be used as a type in `isinstance`.